modules/nw-metallb-configure-return-traffic-proc.adoc

```diff
@@ -0,0 +1,187 @@
+
+.. Apply the configuration for the BGP peer by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f frr-via-vrf.yaml
+----
+
+. Create an `IPAddressPool` CR.
+
+.. Create a file, such as `first-pool.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: first-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 192.169.10.0/32
+----
+
+.. Apply the configuration for the IP address pool by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-pool.yaml
+----
+
+. Create a `BGPAdvertisement` CR.
+
+.. Create a file, such as `first-adv.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: first-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+    - first-pool
+  peers:
+    - frrviavrf <1>
+  nodeSelectors: 
+    - matchLabels:
+        egress-service.k8s.ovn.org/test-server1: "" <2>
+----
+<1> In this example, MetalLB advertises a range of IP addresses from the `first-pool` IP address pool to the `frrviavrf` BGP peer.
+<2> To use the same return path for the traffic originating from the pod, select the node that handles the `LoadBalancer` service. 
+
+.. Apply the configuration for the BGP advertisement by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-adv.yaml
+----
+
+. Create an `EgressService` CR. 
+
+.. Create a file, such as `egress-service.yaml`, with content like the following example:
++
+[source,yaml,options="nowrap",role="white-space-pre"]
+----
+apiVersion: k8s.ovn.org/v1
+kind: EgressService
+metadata:
+  name: server1 <1>
+  namespace: test <2>
+spec:
+  sourceIPBy: "LoadBalancerIP" <3>
+  nodeSelector:
+    matchLabels:
+      vrf: "true" <4>
+  network: "2" <5>
+----
+<1> Specify the name for the egress service. The name of the `EgressService` resource must match the name of the load-balancer service that you want to modify. 
+<2> Specify the namespace for the egress service. The namespace for the `EgressService` must match the namespace of the load-balancer service that you want to modify. The egress service is namespace scoped.
+<3> Assigns the `LoadBalancer` service's ingress IP address as the source IP address for egress traffic.
+<4> This example specifies the node with the label `vrf: "true"` to handle the service's traffic. When a node is selected, OVN-Kubernetes labels the node in the following format `egress-service.k8s.ovn.org/<svc-namespace>-<svc-name>: ""`.
+<5> Specify the routing table for egress traffic.
+
+.Verification
+
+. Verify that you can access the application endpoint of the pods running behind the MetalLB service by running the following command, updating the variables to suit your application endpoint:
++
```

Comment:
This is true, provided that the traffic will come from the isolated network we are setting up.
Also, to verify the srcBy: LoadBalancer part is tricky: the pod should try to reach out a service running outside, and the service should either print the ip the connection is coming from or the user should run tcpdump to see the src ip and make sure it's not the node one but the loadbalancer one.

Reply:
I'd probably keep this simpler and more generic instead of providing the exact commands.